internal/services/v1/schema.go

```diff
@@ -123,13 +123,19 @@ func (ss *schemaServer) WriteSchema(ctx context.Context, in *v1.WriteSchemaReque
 		}
 
 		// For each definition, perform a diff and ensure the changes will not result in any
-		// relationships left without associated schema.
+		// breaking changes.
+		nsDefsWithChanges := make([]*core.NamespaceDefinition, 0, len(nsdefs))
 		for _, nsdef := range nsdefs {
-			if err := shared.SanityCheckExistingRelationships(ctx, rwt, nsdef, existingDefMap); err != nil {
+			diff, err := shared.SanityCheckNamespaceChanges(ctx, rwt, nsdef, existingDefMap)
+			if err != nil {
 				return err
 			}
+
+			if len(diff.Deltas()) > 0 {
+				nsDefsWithChanges = append(nsDefsWithChanges, nsdef)
+			}
 		}
-		log.Ctx(ctx).Trace().Interface("namespaceDefinitions", nsdefs).Msg("validated namespace definitions")
+		log.Ctx(ctx).Trace().Interface("namespaceDefinitions", nsdefs).Interface("namespaceDefinitionsWithChanges", nsDefsWithChanges).Msg("validated namespace definitions")
```

Comment:
wouldn't that be a potentially large payload for a trace?

Reply:
Yes, but it is a trace, so better to include the information than not IMO. Trace should only ever be on if you're debugging something